gateway/ln-gateway/src/gateway_module_v2/mod.rs

```diff
@@ -256,13 +240,30 @@ impl GatewayClientModuleV2 {
             .ok_or(anyhow!("The outgoing contract has not yet been confirmed"))?
             .saturating_sub(EXPIRATION_DELTA_MINIMUM_V2);
 
+        let (payment_hash, amount) = match &payload.invoice_data {
+            InvoiceData::Bolt11(invoice, amount) => (invoice.payment_hash(), amount),
+        };
+
+        ensure!(
+            payment_hash == &payload.contract.payment_hash,
+            "The invoices payment hash does not match the contracts payment hash"
+        );
+
+        let min_contract_amount = self
+            .gateway
+            .routing_info_v2(&payload.federation_id)
+            .await
+            .ok_or(anyhow!("Routing Info not available"))?
+            .send_fee_minimum
+            .add_fee(amount.msats);
+
         let send_sm = GatewayClientStateMachinesV2::Send(SendStateMachine {
             common: SendSMCommon {
                 operation_id,
                 contract: payload.contract.clone(),
                 max_delay,
                 min_contract_amount,
-                invoice: Invoice::Bolt11(payload.invoice),
+                invoice_data: payload.invoice_data,
```

Comment:
nit: as @dpc commonly says, "data" doesn't really add much here since almost everything is data. I think `invoice` or `ln_invoice` works just fine here

Reply:
Done.